includes/data-stores/class-wc-data-store-wp.php

```diff
@@ -243,39 +243,181 @@ protected function get_wp_query_args( $query_vars ) {
 	}
 
 	/**
-	 * Get a valid date for use in WP_Query date queries.
-	 *
-	 * @since 3.1.0
-	 * @param mixed $query_var Value from a WC_Object_Query's query variable.
+	 * Map a valid date query var to WP_Query arguments.
+	 * Valid date formats: YYYY-MM-DD or timestamp, possibly combined with an operator from $valid_operators.
+	 * Also accepts a WC_DateTime object.
+	 * @param mixed $query_var A valid date format
+	 * @param string $key meta or db column key
+	 * @param array $wp_query_args WP_Query args
+	 * @return array Modified $wp_query_args
 	 */
-	protected function get_date_for_wp_query( $query_var, $as_timestamp = false ) {
-		if ( ! $query_var ) {
-			return '';
-		}
+	protected function parse_date_for_wp_query( $query_var, $key, $wp_query_args = array() ) {
+		$query_parse_regex = '/([^.<>]*)(>=|<=|>|<|\.\.\.)([^.<>]+)/';
+		$valid_operators = array( '>', '>=', '=', '<=', '<', '...' );
+
+		// YYYY-MM-DD queries have 'day' precision. Timestamp/WC_DateTime queries have 'second' precision.
+		$precision = 'second';
+
+		$start_date = null;
+		$operator = '=';
+		$end_date = null;
 
+		// Specific time query with a WC_DateTime.
 		if ( is_a( $query_var, 'WC_DateTime' ) ) {
-			$datetime = $query_var;
+			$end_date = $query_var;
+
+		// Specific time query with a timestamp.
 		} elseif ( is_numeric( $query_var ) ) {
-			// Timestamps are handled as UTC timestamps in all cases.
-			$datetime = new WC_DateTime( "@{$value}", new DateTimeZone( 'UTC' ) );
+			$end_date = new WC_DateTime( "@{$query_var}", new DateTimeZone( 'UTC' ) );
+
+		// Query with operators and possible range of dates.
+		} elseif ( preg_match( $query_parse_regex, $query_var, $sections ) ) {
+			if ( ! empty( $sections[1] ) ) {
+				$start_date = is_numeric( $sections[1] ) ? new WC_DateTime( "@{$sections[1]}", new DateTimeZone( 'UTC' ) ) : wc_string_to_datetime( $sections[1] );
```

Comment:
Like our handling in set_date_prop how about we treat timestamp as UTC but string as WP timezone for consistency.